### PR TITLE
libcephfs: add unmount function in cephfs.pyx

### PR DESCRIPTION
**Pull request by renhwztetecs**
we need unmount function when change the mount point

Signed-off-by: huanwen ren <ren.huanwen@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
The failure is true:

```python
Error compiling Cython file:
------------------------------------------------------------
...
            raise make_ex(ret, "error calling ceph_mount")
        self.state = "mounted"

    def unmount(self):
        if self.state == "mounted":
        with nogil:
       ^
------------------------------------------------------------

cephfs.pyx:485:8: Expected an increase in indentation level
Compiling cephfs.pyx because it changed.
Cythonizing cephfs.pyx
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/cephfs/setup.py", line 189, in <module>
    os.path.join(os.path.dirname(__file__), "..", "rados")
  File "/usr/lib/python2.7/dist-packages/Cython/Build/Dependencies.py", line 798, in cythonize
    cythonize_one(*args[1:])
  File "/usr/lib/python2.7/dist-packages/Cython/Build/Dependencies.py", line 915, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: cephfs.pyx
make[3]: *** [src/pybind/cephfs/CMakeFiles/cython_cephfs] Error 1
make[2]: *** [src/pybind/cephfs/CMakeFiles/cython_cephfs.dir/all] Error 2

File: src/test/pybind/test_cephfs.py [hunk 1]
```diff
@@ -11,6 +11,11 @@ def setup_module():
     cephfs = libcephfs.LibCephFS(conffile='')
     cephfs.mount()
 
+def release_module():
+    global cephfs
+    cephfs = libcephfs.LibCephFS(conffile='')
+    cephfs.unmount()
```
Comment (jcsp) [inline src/test/pybind/test_cephfs.py]: Surely you don't mean to construct `cephfs` and then call unmount on it immediately?

**Comment by jcsp** [comment]
The "add unmount test" commit doesn't really seem to add a test, it just calls unmount at the end of existing test, right?  Please could you add a test that mounts, does something, unmounts, then mounts again (using the same LibCephFS instance) and shows that the LibCephFS instance still works properly after the second mount.

File: src/test/pybind/test_cephfs.py (hunk 1 above)
Comment (renhwztetecs) [inline src/test/pybind/test_cephfs.py]: yeah，I think I understand the mistake before

**Comment by renhwztetecs** [comment]
add test_mount_unmount test

File: src/test/pybind/test_cephfs.py [hunk 2]
```diff
@@ -206,3 +206,15 @@ def test_flock():
...
+@with_setup(setup_test)
+def test_mount_unmount():
+    # has been mount in setup_module() 
+    cephfs.unmount()
+    cephfs.mount()
+    test_directory()
+    cephfs.unmount()
+    cephfs.mount()
+    test_open()    
+
```
Comment (ajarr) [inline src/test/pybind/test_cephfs.py]: Please remove blank lines.

File: src/test/pybind/test_cephfs.py [hunk 3]
```diff
@@ -206,3 +206,15 @@ def test_flock():
     cephfs.close(fd2)
 
     cephfs.close(fd)
+
+@with_setup(setup_test)
+def test_mount_unmount():
+    # has been mount in setup_module() 
+    cephfs.unmount()
+    cephfs.mount()
```
Comment (ajarr) [inline src/test/pybind/test_cephfs.py]: Do we need lines 212, 213 and 214?

File: src/test/pybind/test_cephfs.py [hunk 4]
```diff
@@ -206,3 +206,15 @@ def test_flock():
...
+    test_directory()
+    cephfs.unmount()
+    cephfs.mount()
+    test_open()
```
Comment (ajarr) [inline src/test/pybind/test_cephfs.py]: This line has trailing whitespace?

File: src/test/pybind/test_cephfs.py [hunk 5]
```diff
@@ -206,3 +206,15 @@ def test_flock():
     cephfs.close(fd2)
 
     cephfs.close(fd)
+
+@with_setup(setup_test)
+def test_mount_unmount():
+    # has been mount in setup_module()
```
Comment (ajarr) [inline src/test/pybind/test_cephfs.py]: This line has trailing whitespace?

**Comment by ajarr** [comment]
@renhwztetecs , besides the comments looks OK.

**Comment by renhwztetecs** [comment]
@aakso 
all done,thanks

File: src/test/pybind/test_cephfs.py (hunk 3 above)
Comment (renhwztetecs) [inline src/test/pybind/test_cephfs.py]: Looks like it's really redundant.

File: src/test/pybind/test_cephfs.py [hunk 6]
```diff
@@ -206,3 +206,10 @@ def test_flock():
     cephfs.close(fd2)
 
     cephfs.close(fd)
+
+@with_setup(setup_test)
+def test_mount_unmount():
+    test_directory()
+    cephfs.unmount()
+    cephfs.mount()
+    test_open()
```
Comment (ajarr) [inline src/test/pybind/test_cephfs.py]: The line still has a trailing whitespace.

**Comment by ajarr** [comment]
The Jenkins has failed due to,
 
 Could NOT find Python3Interp: Found unsuitable version "2.7.5", but
  required is at least "3" (found /usr/bin/python)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:313 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindPython3Interp.cmake:146 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/CMakeLists.txt:203 (find_package)

as already noted in http://www.spinics.net/lists/ceph-devel/msg32397.html

I tested the PR locally. The included test ran successfully.
http://pastebin.com/t1tNubvW

@jcsp, the test looks OK to you?

File: src/test/pybind/test_cephfs.py (hunk 6 above)
Comment (renhwztetecs) [inline src/test/pybind/test_cephfs.py]: I change it.

**Comment by ajarr** [comment]
LGTM

**Pull request merged**